src/editor/lib/entity.js

```diff
@@ -155,21 +155,25 @@ export function cloneEntityImpl(entity, newId = undefined) {
     { once: true }
   );
 
+  // In cloneEntityImpl
   if (newId) {
     clone.id = newId;
+  } else if (entity.id) {
+    clone.id =
+      entity.id.length === 21 ? createUniqueId() : getUniqueId(entity.id);
   } else {
-    if (entity.id) {
-      if (entity.id.length === 21) {
-        // nanoid generated id, create a new one
-        clone.id = createUniqueId();
-      } else {
-        // Get a valid unique ID for the entity
-        clone.id = getUniqueId(entity.id);
-      }
-    } else {
-      entity.id = createUniqueId();
```

Comment:
huh, you're right, this should have been `clone` instead of `entity` here, bad code on my part :) We currently never go to this branch because we set an id in `EntityCloneCommand` constructor.